src/formats/onenote.ts

```diff
@@ -142,150 +147,197 @@ export class OneNoteImporter extends FormatImporter {
-			progress.reportProgress(0, remainingSections);
-			remainingSections--;
-
+		for (let sectionId of this.selectedIds) {
+			progress.reportProgress(0, remainingProgress);
+			remainingProgress--;
 			let pageCount: number = 0;
 
-			let sectionFolder: TFolder;
-			if (section.parentSectionGroup) {
-				let sectionGroupFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.parentSectionGroup.displayName);
-				sectionFolder = await this.createFolders(sectionGroupFolder.path + '/' + section.displayName);
-			}
-			else sectionFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
-
-			const pagesUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`;
-			let pages: OnenotePage[] = (await this.fetchResource(pagesUrl, 'json')).value;
+			const baseUrl = `https://graph.microsoft.com/v1.0/me/onenote/sections/${sectionId}/pages`;
+			const params = new URLSearchParams({
+			  $select: 'id,title,createdDateTime,lastModifiedDateTime,level,order,contentUrl',
+			  $orderby: 'order',
+			  pagelevel: 'true'
+			});
 
+			const pagesUrl = `${baseUrl}?${params.toString()}`;
+	
+			let pages: OnenotePage[] = ((await this.fetchResource(pagesUrl, 'json')).value);
+			this.insertPagesToSection(pages, sectionId);
+			
 			progress.reportProgress(0, pages.length);
 
 			for (let i = 0; i < pages.length; i++) {
 				const page = pages[i];
-
+				if (!page.title) page.title = `Untitled-${moment().format('YYYYMMDDHHmmss')}`;
 				try {
 					pageCount++;
-					progress.status(`Importing note ${page.title || 'Untitled'}`);
+					progress.status(`Importing note ${page.title}`);
 
 					// Every 50 items, do a few second break to prevent rate limiting
 					if (i !== 0 && i % 50 === 0) {
-						await new Promise(resolve => setTimeout(resolve, 5000));
+						await new Promise(resolve => setTimeout(resolve, 7500));
 					}
-					this.processFile(progress,
-						sectionFolder,
-						outputFolder,
-						await this.fetchResource(`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`, 'text')
-						, page);
 
+					this.processFile(progress,
+						await this.fetchResource(`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`, 'text'),
+						page);
 					progress.reportProgress(pageCount, pages.length);
-
 				}
 				catch (e) {
-					progress.reportFailed(page.title || 'Untitled note', e.toString());
+					progress.reportFailed(page.title, e.toString());
 				}
 			}
 		}
 	}
 
-	async processFile(progress: ImportContext, sectionFolder: TFolder, outputFolder: TFolder, content: string, page: OnenotePage) {
-		try {
-			const splitContent = this.convertFormat(content);
-
-			if (this.importDrawingsOnly) {
-				// TODO, when InkML support is added
+	insertPagesToSection(pages: OnenotePage[], sectionId: string, parentEntity?: OnenoteEntityHierarchyModel | undefined) {
```

Comment:
Similar to the previous comment, please update the signature of this method and remove the type assertions from this method.

```
insertPagesToSection(pages: OnenotePage[], sectionId: string, parentEntity?: Notebook | SectionGroup) {
```